### PR TITLE
fix: radio and checkbox properties and removed reakit usages

### PR DESCRIPTION
**Pull request by matthprost**
## Type

- Enhancement

#### What is expected?

I remove most of `aria-checked` that were not useful + I removed reakit. This impact `SelectableCard` and `SwitchButton`.

### CODE REVIEW

**Review by chambo-e — APPROVED** [review]

File: src/components/Checkbox/index.tsx [hunk 1]
```diff
@@ -185,7 +176,11 @@ type CheckboxProps = Pick<
   checked?: boolean | 'indeterminate'
   className?: string
   ['data-visibility']?: string
-} & Required<Pick<ReakitCheckboxProps, 'onChange'>>
+} & Required<Pick<InputHTMLAttributes<HTMLInputElement>, 'onChange'>> &
```
Comment (DorianMaliszewski) [inline src/components/Checkbox/index.tsx:179]: Pas obligé de le mettre en required non ?

File: src/components/Checkbox/index.tsx [hunk 2]
```diff
@@ -208,7 +203,7 @@ const Checkbox = forwardRef(
     ref: ForwardedRef<HTMLInputElement>,
   ) => {
     const hasChildren = !!children
-    const { state, setState } = useCheckboxState({ state: checked })
+    const [state, setState] = useState<boolean | 'indeterminate'>(checked)
```
Comment (DorianMaliszewski) [inline src/components/Checkbox/index.tsx:206]: Should we remove this behaviour ? This component doesn't need to handle its own state. The user should use checked prop and onChange to change it ?

File: src/components/Checkbox/index.tsx [hunk 3]
```diff
@@ -247,20 +239,19 @@ const Checkbox = forwardRef(
...
-          <StyledReakitCheckbox
+          <CheckboxInput
+            type="checkbox"
             aria-invalid={!!error}
             aria-describedby={error ? `${computedName}-hint` : undefined}
-            aria-checked={
-              state === 'indeterminate' ? 'mixed' : (state as boolean)
-            }
-            checked={state === 'indeterminate' ? false : (state as boolean)}
+            aria-checked={state === 'indeterminate' ? 'mixed' : undefined}
```
Comment (QuiiBz) [inline src/components/Checkbox/index.tsx:253]: I'm not sure about this, but I think it should behave the same as `checked`?
```suggestion
            aria-checked={state === 'indeterminate' ? 'mixed' : state}
```

Comment (matthprost) [reply src/components/Checkbox/index.tsx:253]: @QuiiBz the best is not to use `aria-checked` as it is for a11y. In that case we need it only for `mixed` state as it is the only way to differentiate for true/false state :)

**Review by QuiiBz — APPROVED** [review]

**Pull request merged**